src/data/equiv/fin.lean

```diff
@@ -307,6 +307,22 @@ def fin_prod_fin_equiv : fin m × fin n ≃ fin (m * n) :=
         ... = y.1 : nat.mod_eq_of_lt y.2),
   right_inv := λ x, fin.eq_of_veq $ nat.mod_add_div _ _ }
 
+/-- Promote a `fin n` into a larger `fin m`, as a subtype where the underlying
+values are retained. This is the `order_iso` version of `fin.cast_le`. -/
+def fin.cast_le_order_iso {n m : ℕ} (h : n ≤ m) : fin n ≃o {i : fin m // (i : ℕ) < n} :=
```

Comment:
Does this generate the right lemmas?
```suggestion
@[simps apply symm_apply]
def fin.cast_le_order_iso {n m : ℕ} (h : n ≤ m) : fin n ≃o {i : fin m // (i : ℕ) < n} :=
```

Reply:
I don't think so, because they aren't in simp-normal form. They put `fin.cast_le h` on the RHS.

Reply:
How about with `simp_rhs := tt`?

Reply:
You still get
```
fin.cast_le_order_iso_apply :
  ∀ {n m : ℕ} (h : n ≤ m) (i : fin n), ⇑(fin.cast_le_order_iso h) i = ⟨⇑(fin.cast_le h) i, _⟩
```

Reply:
That looks like a pretty reasonable statement to me. We don't have a simp lemma that unfolds `fin.cast_le` into subtype.mk, so it probably is sensible to be consistent with this version.